src/renderer/runner.ts

```diff
@@ -333,7 +333,7 @@ export class Runner {
   private buildChildEnvVars(): { [x: string]: string | undefined } {
     const { isEnablingElectronLogging, environmentVariables } = this.appState;
 
-    const env = { ...process.env };
+    const env: Record<string, string> = {};
```

Comment:
The `else` block on lines 342-345 can be removed, there's nothing to `delete` anymore

Reply:
Good catch, didn't notice those lines. That logic should actually probably be shifted to the main process so that the behavior matches what it was before the refactoring. Otherwise if those variables are in the environment in general, there will be a behavior change from before the main process refactor.

I'll look at refactoring this PR.